x/upgrade/keeper/keeper.go

```diff
@@ -140,16 +152,22 @@ func (k Keeper) SetModuleVersionMap(ctx sdk.Context, vm module.VersionMap) {
 			versionStore.Set(nameBytes, verBytes)
 		}
 	}
+
+	return nil
 }
 
 // GetModuleVersionMap returns a map of key module name and value module consensus version
 // as defined in ADR-041.
-func (k Keeper) GetModuleVersionMap(ctx sdk.Context) module.VersionMap {
-	store := ctx.KVStore(k.storeKey)
-	it := storetypes.KVStorePrefixIterator(store, []byte{types.VersionMapByte})
+func (k Keeper) GetModuleVersionMap(ctx context.Context) (module.VersionMap, error) {
+	store := k.storeService.OpenKVStore(ctx)
+	prefix := []byte{types.VersionMapByte}
+	it, err := store.Iterator(prefix, storetypes.PrefixEndBytes(prefix))
+	if err != nil {
+		return nil, err
```

Comment:
I think we need to close the iterator before returning the error
```suggestion
		_ = it.Close()
		return nil, err
```